app/src/main/java/com/github/polypoly/app/network/RemoteDB.kt

```diff
@@ -0,0 +1,159 @@
+package com.github.polypoly.app.network
+
+import android.util.Log
+import com.github.polypoly.app.game.Skin
+import com.github.polypoly.app.game.User
+import com.github.polypoly.app.global.Settings.Companion.DB_ALL_USERS_ID_PATH
+import com.github.polypoly.app.global.Settings.Companion.DB_USERS_PROFILES_PATH
+import com.github.polypoly.app.global.Settings.Companion.DB_USER_BIO_DIRECTORY
+import com.github.polypoly.app.global.Settings.Companion.DB_USER_NAME_DIRECTORY
+import com.github.polypoly.app.global.Settings.Companion.DB_USER_SKIN_DIRECTORY
+import com.github.polypoly.app.global.Settings.Companion.DB_USER_STATS_DIRECTORY
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ktx.getValue
+import java.util.concurrent.CompletableFuture
+
+open class RemoteDB(
+    private val db: FirebaseDatabase?
+) : IRemoteStorage {
+
+    private lateinit var rootRef: DatabaseReference
+    private lateinit var usersProfilesRef: DatabaseReference
+
+    init {
+        if (db != null) {
+            rootRef = db.getReference()
```

Comment:
kotlin allows to use db.reference directly

Reply:
This is now needed as we specify a root path to differentiate between the "live" and "test" parts of the db